game/tournament.py

```diff
@@ -36,22 +44,53 @@ def get_results_table(self):
     def get_scores(self):
         return self._scores_dict
 
+    def timeout_handler(self, arg1, arg2):
+        raise TimeLimitReached
+
     def run(self):
         for match in self._matches:
-            match.play_full_match()
+            abort_current_match = False
+            penalized_player = None
+            signal.signal(signal.SIGALRM, self.timeout_handler)
+            for player in match.get_players():
+                signal.setitimer(signal.ITIMER_REAL,self._prepare_time_limit)
```

Comment:
I never used this approach, so if you think this is the way to go I'm happy with it. Having said that, doing just:
```
start_time = time.time()
player.prepare()
end_time = time.time()
if (end_time - start_time) > self._prepare_time_limit:
    ....
```
seemed like an easier / more reliable approach. But again, if you researched on this I'm happy to go this route.